router-tests/structured_logging_test.go

```diff
@@ -1567,20 +1637,22 @@ func TestAccessLogs(t *testing.T) {
 
 				productContext := requestLog.All()[1].ContextMap()
 				productSubgraphVals := map[string]interface{}{
-					"log_type":         "client/subgraph",
-					"subgraph_name":    "products",
-					"subgraph_id":      "3",
-					"status":           int64(403),
-					"method":           "POST",
-					"path":             "/graphql",
-					"query":            "", // http query is empty
-					"ip":               "[REDACTED]",
-					"service_name":     "service-name",                                                     // From request header
-					"error":            "Failed to fetch from Subgraph 'products' at Path: 'employees'.",   // From context
-					"operation_hash":   "16884868987896027258",                                             // From context
-					"operation_sha256": "049efe2ebbdf2e4845e69f69cb7965963b118612a6247ab6d91b1961ea0158dc", // From context
-					"operation_name":   "employees",                                                        // From context
-					"operation_type":   "query",                                                            // From context
+					"log_type":               "client/subgraph",
+					"subgraph_name":          "products",
+					"subgraph_id":            "3",
+					"status":                 int64(403),
+					"method":                 "POST",
+					"path":                   "/graphql",
+					"query":                  "", // http query is empty
+					"ip":                     "[REDACTED]",
+					"error":                  "Failed to fetch from Subgraph 'products' at Path: 'employees'.",   // Automatically added from error
```

Comment:
Looking at the normal access logs, it seems like we only automatically add `error` when we encounter a panic issue, but not a normal graphql error. Should I not be adding `error` automatically for subgraphs, and keep it restricted to the `response_error_message` and `request_error` context fields they can add?

Reply:
That's a great observation. For now, I recommend to follow your proposal and reserve `error` for a different use cases.